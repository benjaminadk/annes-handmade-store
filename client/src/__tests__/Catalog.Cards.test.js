import React from 'react'
import Cards from '../components/Catalog.Cards'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'

const products = [
  {
    id: 1,
    title: 'Test Product',
    images: [
      'https://shopping-site.s3.amazonaws.com/images/red-bracelet-starfish-1-opt'
    ],
    price: 20
  }
]

describe('Catalog Cards Component', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Cards />).toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('receives array of products as a prop', () => {
    const wrapper = shallow(<Cards products={products} />)
    expect(wrapper.prop('products').length).toEqual(1)
  })
})
