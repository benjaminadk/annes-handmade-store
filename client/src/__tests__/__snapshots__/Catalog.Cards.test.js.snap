// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Catalog Cards Component renders correctly 1`] = `
<div
  className="CatalogCards-root-1"
/>
`;
